argilla-server/src/argilla_server/telemetry.py

```diff
@@ -17,114 +17,248 @@
 from argilla_server.constants import DEFAULT_USERNAME
-from argilla_server.models import User
+from argilla_server.models import (
+    Dataset,
+    Field,
+    FloatMetadataPropertySettings,
+    IntegerMetadataPropertySettings,
+    MetadataProperty,
+    MetadataPropertySettings,
+    Question,
+    Record,
+    TermsMetadataPropertySettings,
+    User,
+    VectorSettings,
+    Workspace,
+)
 from argilla_server.settings import settings
 from argilla_server.utils._telemetry import (
     is_running_on_docker_container,
     server_deployment_type,
 )
 
-try:
-    from analytics import Client  # This import works only for version 2.2.0
-except (ImportError, ModuleNotFoundError):
-    # TODO: show some warning info
-    settings.enable_telemetry = False
-    Client = None
-
-
 _LOGGER = logging.getLogger(__name__)
 
 
 @dataclasses.dataclass
 class TelemetryClient:
     enable_telemetry: dataclasses.InitVar[bool] = settings.enable_telemetry
-    disable_send: dataclasses.InitVar[bool] = False
-    api_key: dataclasses.InitVar[str] = settings.telemetry_key
-    host: dataclasses.InitVar[str] = "https://api.segment.io"
 
     _server_id: Optional[uuid.UUID] = dataclasses.field(init=False, default=None)
 
     @property
     def server_id(self) -> uuid.UUID:
         return self._server_id
 
-    def __post_init__(self, enable_telemetry: bool, disable_send: bool, api_key: str, host: str):
-        from argilla_server._version import __version__
-
+    def __post_init__(self, enable_telemetry: bool):
         self._server_id = uuid.UUID(int=uuid.getnode())
         self._system_info = {
+            "server_id": str(self._server_id),
             "system": platform.system(),
             "machine": platform.machine(),
             "platform": platform.platform(),
-            "python_version": platform.python_version(),
             "sys_version": platform.version(),
             "deployment": server_deployment_type(),
             "docker": is_running_on_docker_container(),
-            "version": __version__,
         }
 
         _LOGGER.info("System Info:")
         _LOGGER.info(f"Server id: {self.server_id}")
         _LOGGER.info(f"Context: {json.dumps(self._system_info, indent=2)}")
+        self.enable_telemetry = enable_telemetry
+
+    @staticmethod
+    def _process_request_info(request: Request):
+        return {header: request.headers.get(header) for header in ["user-agent", "accept-language"]}
+
+    @staticmethod
+    def _process_workspace_model(workspace: Workspace):
+        return {
+            "workspace_id": str(workspace.id),
+            "workspace_hash": str(uuid.uuid5(namespace=_TELEMETRY_CLIENT.server_id, name=workspace.name)),
+        }
+
+    @staticmethod
+    def _process_dataset_model(dataset: Dataset):
+        return {
+            "dataset_id": str(dataset.id),
+            "dataset_hash": str(uuid.uuid5(namespace=_TELEMETRY_CLIENT.server_id, name=dataset.name)),
```

Comment:
@frascuchon was there a reason to include these hashed? otherwise I would opt for just storing the ID

Reply:
I agree, we can simplify and just reason based on IDs.